### DISCUSSION TITLE
component with <script client:load> slows down build 10X

### DISCUSSION THREAD

**Discussion by joesonw** [ElMassimo/iles#233] [General]
library versions: iles v0.8.7 vite v3.2.4 
hardware: apple m1 chip

I have around 200 pages to generate, it takes about 30s to build. after I reference this component that has client load script, it takes 300+ seconds to build. (I tried move it to layout, of course it helped, but the code became harder to maintain, since the logic and view are separate now).

Is this situation normal?

**Comment by ElMassimo** [comment]
Hi Qiaosen!

That behavior seems unusual, it could be related to having many islands in each of those 200 pages.

Can you share the build stats? It should output [how much time is spent](https://github.com/ElMassimo/iles/blob/2dae8a321f4e027e3b5769c9753054264b43b77e/packages/iles/src/node/build/build.ts#L26-L27) in the [`bundleIslands` phase](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/build/islands.ts#L20-L26) of the build.

It would be interesting to know if you are passing different props to those islands in each page, perhaps this is causing Rollup to analyze them differently.

There's plenty of room for optimization at the moment, as similar usages of islands could be de-duplicated before passing them to Rollup. So far, I've assumed that Rollup would be able to deal with duplication efficiently, but that might not be the case. Example: if many pages render the same island in the header, passing "a single island" to Rollup would surely speed up the build.

**Reply by joesonw** [reply]
I can confirm`building islands bundle` is where the time difference is.

Yeah, after commenting out codes, I located it to be client load scripts. This component is located out side of iles project src folder. And it is used by every page.
